indico/web/client/js/custom_elements/ind_datepicker.js

```diff
@@ -0,0 +1,631 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import './ind_datepicker.scss';
+
+const TODAY = (function() {
+  const now = new Date();
+  now.setHours(0, 0, 0, 0);
+  return now;
+})();
+const DEFAULT_LOCALE =
+  Intl.DateTimeFormat?.resolvedOptions?.().locale ||
+  document.documentElement.dataset.canonicalLocale ||
+  document.documentElement.lang ||
+  navigator.language;
```

Comment:
Is it really a good idea to get this from the browser and not from what Indico considers the current language? I think it'd be weird if someone uses e.g. English Indico on a French browser, and suddenly get French-style date formatting.

Especially since the regform date field in Indico has an option to configure the format...

Reply:
Ah, yeah, I knew I was forgetting something. I need to add the locale of the page as the `data-canonical-locale` to the `<html>` element in the template and remove the rest of the options. Good catch, thanks.

Reply:
I'd get it by calling `moment.locale()` which should already be set on all pages. That way we don't introduce another source of truth for the current locale...